psi4/driver/procrouting/response/scf_response.py

```diff
@@ -534,54 +559,68 @@ def tdscf_excitations(wfn,
-            R_velocity, spin_mult) in enumerate(_results):
-        sym_descr = f"{irrep_GS}->{irrep_ES} ({1 if spin_mult== 'singlet' else 3} {irrep_trans})"
+    for i, x in enumerate(_results):
+        sym_descr = f"{x.irrep_GS}->{x.irrep_ES} ({1 if x.spin_mult== 'singlet' else 3} {x.irrep_trans})"
 
-        E_ex_ev = constants.conversion_factor('hartree', 'eV') * E_ex_au
+        E_ex_ev = constants.conversion_factor('hartree', 'eV') * x.E_ex_au
 
-        E_tot_au = wfn.energy() + E_ex_au
+        E_tot_au = wfn.energy() + x.E_ex_au
 
         # prepare return dictionary for this root
         solver_results.append({
-            "EXCITATION ENERGY": E_ex_au,
-            "LENGTH-GAUGE ELECTRIC DIPOLE TRANSITION MOMENT": edtm_length,
-            "LENGTH-GAUGE OSCILLATOR STRENGTH": f_length,
-            "VELOCITY-GAUGE ELECTRIC DIPOLE TRANSITION MOMENT": edtm_velocity,
-            "VELOCITY-GAUGE OSCILLATOR STRENGTH": f_velocity,
-            "MAGNETIC DIPOLE TRANSITION MOMENT": mdtm,
-            "LENGTH-GAUGE ROTATORY STRENGTH": R_length,
-            "VELOCITY-GAUGE ROTATORY STRENGTH": R_velocity,
-            "SYMMETRY": irrep_trans,
-            "SPIN": spin_mult,
+            "EXCITATION ENERGY": x.E_ex_au,
+            "LENGTH-GAUGE ELECTRIC DIPOLE TRANSITION MOMENT": x.edtm_length,
+            "LENGTH-GAUGE OSCILLATOR STRENGTH": x.f_length,
+            "VELOCITY-GAUGE ELECTRIC DIPOLE TRANSITION MOMENT": x.edtm_velocity,
+            "VELOCITY-GAUGE OSCILLATOR STRENGTH": x.f_velocity,
+            "MAGNETIC DIPOLE TRANSITION MOMENT": x.mdtm,
+            "LENGTH-GAUGE ROTATORY STRENGTH": x.R_length,
+            "VELOCITY-GAUGE ROTATORY STRENGTH": x.R_velocity,
+            "SYMMETRY": x.irrep_trans,
+            "SPIN": x.spin_mult,
+            "RIGHT EIGENVECTOR ALPHA": x.R_eigvec if restricted else x.R_eigvec[0],
+            "LEFT EIGENVECTOR ALPHA": x.L_eigvec if restricted else x.L_eigvec[0],
+            "RIGHT EIGENVECTOR BETA": x.R_eigvec if restricted else x.R_eigvec[1],
+            "LEFT EIGENVECTOR BETA": x.L_eigvec if restricted else x.L_eigvec[1],
         })
 
         # stash in psivars/wfnvars
         ssuper_name = wfn.functional().name()
-        wfn.set_variable(f"TD-{ssuper_name} ROOT {i+1} TOTAL ENERGY - {irrep_ES} SYMMETRY", E_tot_au)
-        wfn.set_variable(f"TD-{ssuper_name} ROOT 0 -> ROOT {i+1} EXCITATION ENERGY - {irrep_ES} SYMMETRY", E_ex_au)
+        wfn.set_variable(f"TD-{ssuper_name} ROOT {i+1} TOTAL ENERGY - {x.irrep_ES} SYMMETRY", E_tot_au)
+        wfn.set_variable(f"TD-{ssuper_name} ROOT 0 -> ROOT {i+1} EXCITATION ENERGY - {x.irrep_ES} SYMMETRY", x.E_ex_au)
         wfn.set_variable(
-            f"TD-{ssuper_name} ROOT 0 -> ROOT {i+1} LENGTH-GAUGE OSCILLATOR STRENGTH - {irrep_ES} SYMMETRY", f_length)
+            f"TD-{ssuper_name} ROOT 0 -> ROOT {i+1} LENGTH-GAUGE OSCILLATOR STRENGTH - {x.irrep_ES} SYMMETRY",
+            x.f_length)
```

Comment:
are all these in atomic units?

Reply:
Everything is in atomic units. I'll be more specific with that.

Reply:
Now I'm wondering, should the units be part of the variables name? Or is it generally assumed that whatever Psi4 returns is in atomic units?